recipes/foxi/all/conanfile.py

```diff
@@ -77,6 +73,16 @@ def package(self):
             rename(self, src=dll_file, dst=os.path.join(self.package_folder, "bin", os.path.basename(dll_file)))
 
     def package_info(self):
-        self.cpp_info.libs = ["foxi_dummy", "foxi_loader"]
+        self.cpp_info.components["header_only"].libs = []
+        self.cpp_info.components["header_only"].libdirs = []
+        self.cpp_info.components["header_only"].set_property("cmake_target_name", "foxi")
+
+        self.cpp_info.components["foxi_loader"].libs = ["foxi_loader"]
+        self.cpp_info.components["foxi_loader"].set_property("cmake_target_name", "foxi_loader")
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.components["foxi_loader"].system_libs = ["dl"]
+
+        self.cpp_info.components["foxi_dummy"].libs = ["foxi_loader"]
+        self.cpp_info.components["foxi_dummy"].set_property("cmake_target_name", "foxi_dummy")
```

Comment:
where do these CMake target names come from?

Reply:
Precisely from upstream foxi CMakeLists.txt. https://github.com/houseroad/foxi/blob/master/CMakeLists.txt#L117

Upstream CMakeLists installs `foxi`, `foxi_loader` and `foxi_dummy` targets. These targets are un-name spaced (no foxi:: anywhere).

Currently foxi::foxi doesn't offer the granularity required to link against only the loader.

Reply:
I don't see an export of these targets upstream, nor a CMake config file.

Reply:
```
install(TARGETS
  foxi foxi_dummy foxi_loader
  EXPORT ONNXTargets DESTINATION lib)
```
I see an export as `ONNXTargets`... but I guess this is dead code from when they forked this project from onnx?

Usually they incorporate thirdparty dependencies as submodules and build them by source without find_package nor Config/Module files in the middle.

* Pytorch uses add_submodule https://github.com/pytorch/pytorch/blob/v1.13.1/cmake/Dependencies.cmake#L1560 and `foxi_loader` target.
* glow not even that, https://github.com/pytorch/glow/blob/d7bd6c59e68a105edafe094ee77c987903eb24a5/lib/Importer/CMakeLists.txt#L41 they just include headers (header-only `foxi`  target would be good for that)